### PR TITLE
fix(authz): move opa out of startup call

### PR DESCRIPTION
**Pull request by strantalis**
This pull request helps address https://github.com/opentdf/platform/issues/1033 and introducing the concept of a mode. 

Currently we are starting `opa` for every service. When in practice it is only consumed by the authorization service that is dependent on executing the `rego` policy.

In this update, I have removed OPA and instead directly eval the Rego policy, which can be either embedded or passed in via configuration. Until we have clear use cases for the additional management functionality that OPA provides, we can focus on just executing our entitlements policy.

### CODE REVIEW

File: service/authorization/authorization.go [hunk 1]
```diff
@@ -57,28 +65,59 @@ func NewRegistration() serviceregistry.Registration {
...
+
 			if err := mapstructure.Decode(srp.Config.ExtraProps, &authZCfg); err != nil {
 				panic(fmt.Errorf("invalid auth svc cfg [%v] %w", srp.Config.ExtraProps, err))
 			}
 
-			config := clientcredentials.Config{ClientID: authZCfg.ClientID, ClientSecret: authZCfg.ClientSecret, TokenURL: authZCfg.TokenEndpoint}
-			slog.Debug("authorization service client config", slog.Any("config", config))
-			newTokenSource := oauth2.ReuseTokenSourceWithExpiry(nil, config.TokenSource(context.Background()), tokenExpiryDelay)
-			slog.Debug("authorization service token source created", slog.Any("token_source", newTokenSource))
+			slog.Debug("authorization service config", slog.Any("config", authZCfg))
```
Comment (jakedoublev) [inline service/authorization/authorization.go:88]: Did we end up deploying a global logging solution for masking config secrets in logs?

File: service/authorization/authorization.go [hunk 2]
```diff
@@ -337,15 +373,8 @@ func (as *AuthorizationService) GetEntitlements(ctx context.Context, req *author
 		}
 		var attributeFqns []string
 		for _, attr := range listAttributeResp.GetAttributes() {
-			ns := attr.GetNamespace().GetName()
-			an := attr.GetName()
 			for _, val := range attr.GetValues() {
```
Comment (jakedoublev) [inline service/authorization/authorization.go:376]: can avoid this extra looping once https://github.com/opentdf/platform/pull/1050 is merged

File: service/authorization/authorization.go (hunk 1 above)
Comment (strantalis) [reply service/authorization/authorization.go:88]: No it was only implemented for the root config object and I don't think its working last time I looked.

Comment (strantalis) [reply service/authorization/authorization.go:91]: https://github.com/opentdf/platform/issues/751

**Review by pflynn-virtru — DISMISSED** [review]

File: service/authorization/authorization.go [hunk 3]
```diff
@@ -23,31 +24,41 @@ import (
...
+	config      Config
 	logger      *logger.Logger
 	tokenSource *oauth2.TokenSource
+	eval        rego.PreparedEvalQuery
 }
 
 type Config struct {
 	// Entity Resolution Service URL
-	ERSURL string `mapstructure:"ersurl"`
+	ERSURL string `mapstructure:"ersurl" default:"http://localhost:8080/entityresolution/resolve"`
```
Comment (jrschumacher) [inline service/authorization/authorization.go:46]: I'm not sure we need to define the default here if the plan is to remove it.

**Review by jakedoublev — APPROVED** [review]

**Pull request merged**